app/qml/components/MMToolbar.qml

```diff
@@ -144,8 +144,19 @@ Rectangle {
       if (btnData.visible) ++visibleButtonsCount
     }
 
+    // Find how many short buttons we can fit into toolbar:
+    // We want to show minimumToolbarButtonsInMainToolbar buttons
+    // if possible, even at the cost that they will be shorter.
+    var maxButtonsInToolbar = Math.floor( toolbar.width / internal.preferredToolbarButtonWidth )
+    if (maxButtonsInToolbar < internal.minimumToolbarButtonsInMainToolbar) {
+      maxButtonsInToolbar = Math.floor( toolbar.width / internal.minimumToolbarButtonWidth )
+      if (maxButtonsInToolbar > internal.minimumToolbarButtonsInMainToolbar) {
+        // we do not want to show more buttons than minimum even they would fit
+        maxButtonsInToolbar = internal.minimumToolbarButtonsInMainToolbar
+      }
+    }
```

Comment:
I was thinking about something like this - would that work? 🤔 

```suggestion
    let maxButtonsInToolbar = Math.floor( toolbar.width / internal.preferredToolbarButtonWidth )
    
    if ( Screen.primaryOrientation === Qt.PortraitOrientation || Qt.InvertedPortraitOrientation ) {
      maxButtonsInToolbar = Math.max( maxButtonsInToolbar, internal.minimumToolbarButtonsInMainToolbar)
    }
```

Reply:
this would work on mobiles , but when you will play with width on desktop it will be screwed... and we have windows production version at minimum